### DISCUSSION TITLE
show and evaluate markdown chunk

### DISCUSSION THREAD

**Discussion by elgabbas** [quarto-dev/quarto-cli#3303] [Q&A] [answered]
Hello,

Is it possible to show and evaluate a markdown text in Quarto / Rmarkdown?

I need to show both the raw markdown script and its compiled (evaluated) form, one after another. In Quarto, I can use the following code to show/format a markdown code, but this does not provide evaluated results (i.e. the link).
```{markdown}
#| echo: true
#| eval: true
[Quarto](https://quarto.org)
```

Please note that I also asked this question on [StackOverflow](https://stackoverflow.com/questions/74335328/), without a solution yet.
Thanks

**Comment by mcanouil** [comment]
Markdown is not a valid language for code chunk, so you'all have to use regular markdown syntax as in Stack Overflow, GitHub and actually any places using markdown, i.e., "The following code `[Quarto](https://quarto.org)` will be formatted as [Quarto](https://quarto.org)".
Where `` "The following code `[Quarto](https://quarto.org)` will be formatted as [Quarto](https://quarto.org)" `` is the actual markdown code.

**Reply by elgabbas** [reply]
Thanks @mcanouil ..
I understand that. However, I want to make a tutorial for teaching/learning Quarto/Rmarkdown. I would like to be able to show the input markdown script along with its output [similar to the use of python or R], without the need to repeat each piece of code twice, one to be evaluated as a code chunk and another to be evaluated as markdown. Is this even possible?

**Reply by mcanouil** [reply]
As I said, "markdown" is not a `knitr` language for computation and far as I know it's not a `Jupyter` one either.

**Reply by mcanouil** [reply]
One thing you can do is generating everything from a computing language, e.g., example below.

If you are happy with this answer, please mark the thread as the answer to let others know it has been answered.

````
---
title: "Untitled"
format: html
engine: knitr
---

```{r}
#| include: false
md <- function(x) {
  cat(
    "\n::: {layout-ncol=2}\n",
    "\n### Code\n",
    "\n```md\n", x, "\n```\n\n\n\n",
    "\n### Output\n",
    x,
    "\n:::\n",
    sep = ""
  )
}
```


```{r}
#| echo: false
#| output: asis
md("[Quarto](https://quarto.org)")
```

````

<img width="752" alt="Sreenshot showing the HTML document from the code above with a title 'Untitled', two colums with on the left code h3 title followed by a markdown chunk, and on the right a h3 title followed by the rendered markdown" src="https://user-images.githubusercontent.com/8896044/201479118-a2f554f6-061f-44a1-8f71-9840f89752e7.png">

**Reply by elgabbas** [reply]
This makes sense... Thanks

**Comment by elgabbas** [comment, marked as answer]
Actually, I was able to find a direct solution to my question. I received a comment from Albert Krewinkel (@tarleb) on my question on StackOverflow. I copy below his [comment](https://stackoverflow.com/questions/74335328/quarto-show-and-evaluate-markdown-chunk/74420450#74420450):

---

> You can build your own evaluator with a Quarto Lua filter:
```
function CodeBlock (cb)
  if cb.classes:includes 'markdown' and cb.classes:includes 'eval' then
    return {cb} .. pandoc.read(cb.text).blocks
  end
end
```
> Save the code to a file in your project and add that file to the filters YAML entry:
```
---
filters:
  - markdown-examples.lua
---
```
> The filter requires slightly different syntax though:
```
```{.markdown .eval}
[Quarto](https://quarto.org)
```

> This isn't fully satisfying though, as it only supports pandoc's Markdown and doesn't know about Quarto extensions to it.

**Reply by mcanouil** [reply]
This is not a direct solution since it still involves a computing language, i.e., you still need to write a code chunk and you need a Lua filter which in the end is no different than writing an R function and calling the markdown in an R code chunk.

Anyway, it's great you found something you find to your own taste. 😉
